src/utils_internal.rs

```diff
@@ -331,7 +331,9 @@ pub async fn build_ble_stream(
                 // Data from user, forward it to the device
                 from_server = server.read(&mut buf) => {
                     let len = from_server.map_err(duplex_write_error_fn)?;
-                    ble_handler.write_to_radio(&buf[..len]).await?;
+                    if len != 0 {
+                        ble_handler.write_to_radio(&buf[..len]).await?;
```

Comment:
I think we can also return `Err(InternalStreamError::ConnectionLost)` in case `len == 0`.

I can look into this later, you can merge this PR as it is an improvement.